internal/controller/config/templates/service.yaml.tmpl

```diff
@@ -12,6 +12,12 @@ metadata:
     app.kubernetes.io/created-by: model-registry-operator
     app.kubernetes.io/part-of: model-registry
     app.kubernetes.io/managed-by: model-registry-operator
+  annotations:
+    {{- if and .Spec.Istio .Spec.Istio.Gateway .Spec.Istio.Gateway.Domain}}
+    routing.opendatahub.io/external-address: {{.Name}}-rest.{{.Spec.Istio.Gateway.Domain}}:{{.Spec.Istio.Gateway.Rest.Port}}
+    routing.opendatahub.io/external-address-rest: {{.Name}}-rest.{{.Spec.Istio.Gateway.Domain}}:{{.Spec.Istio.Gateway.Rest.Port}}
+    routing.opendatahub.io/external-address-grpc: {{.Name}}-grpc.{{.Spec.Istio.Gateway.Domain}}:{{.Spec.Istio.Gateway.Grpc.Port}}
```

Comment:
why not just making the default one to be the rest, so to avoid the `routing.opendatahub.io/external-address-rest` annotation which seems redundant?

Reply:
For uniformity across protocols. And it also makes it clear that the rest endpoint is the default endpoint as well. Does that make sense?

Reply:
The dashboard could just use the `-rest` endpoint if we want to, since it specifically expects the REST API. Not sure what you mean by uniformity across protocols being a reason for the default annotation.

Also, are these "external" in the sense described by this doc or should we be using "public"? (i.e. are they accessible outside the cluster?) https://docs.google.com/document/d/1UA9c8DWVQg1nzkiw3b34xofL8KCw_MM2hD5MOf-P2VU/edit

Reply:
I was thinking that dashboard doesn't care about the grpc protocol, so it would be easier to have a default annotation `routing.opendatahub.io/external-address` for clients like dashboard that always use a _default_/_common_ protocol for the service. 
For other clients, they can look for the endpoint for a specific protocol by adding the protocol suffix. Hence the redundancy. 
I am hesitant about using `public`, since that has RBAC connotations, i.e. whether they are publicly accessible i.e. not secured. `External` is external to the mesh right now.

Reply:
The dashboard does use gRPC for DSP, just not ModelRegistry currently. If we extend this pattern to other services we'll want to have each usage be explicit about what protocol/API it's using.

Reply:
I'll drop the annotation without the protocol suffix in that case.

Reply:
Thanks! Yeah I think that makes sense for our use case